lib/xml_stream_writer.ex

```diff
@@ -17,30 +17,38 @@
 #
 
 defmodule XMLStreamWriter do
+  @spec new_document :: {:ok, String.t(), []}
   def new_document() do
     {:ok, "", []}
```

Comment:
Here, in `start_document` and in `characters` the second element of the output is not an `iolist`. In all other fuctions, the second element of the output is an `iolist`. 
@bettio is it better to have less uniform typespecs (`String.t()` in some parts, `iolist()` elsewhere) or have uniform typespecs by wrapping the `string` inside `iolist`?

Reply:
Use `iodata()`